### PR TITLE
Fix Windows build; add Windows to CI

### PR DESCRIPTION
**Pull request by bgilbert**
While we're here, deduplicate the separate CI jobs into one matrix job.  I did that in the last commit so it can be dropped if desired.

### CODE REVIEW

File: .github/workflows/run_unit_tests.yml [hunk 1]
```diff
@@ -76,3 +76,36 @@ jobs:
...
+      fail-fast: false
+      matrix:
+        os: [windows-latest]
+        cc: [clang]
+    steps:
+    - uses: actions/checkout@v3
+    - name: Install build and test dependencies
+      # Pin to Meson 0.60.1 for
+      # https://github.com/mesonbuild/meson/issues/10022
+      run: pip install meson==0.60.1
```
Comment (bgilbert) [inline .github/workflows/run_unit_tests.yml:50]: Without this pin, I get:

```
subprojects\check-0.15.2\tests\libstaticcheck.a : fatal error LNK1107: invalid or corrupt file: cannot read at 0x2452
```

It appears from https://github.com/mesonbuild/meson/issues/10022 that this was fixed in 0.60.1 but the fix wasn't forward-ported to 0.61+.

cc @eli-schwartz

File: .github/workflows/run_unit_tests.yml [hunk 2]
```diff
@@ -32,47 +39,43 @@ jobs:
           valgrind
         sudo apt-get clean
 
+    - name: Install build and test dependencies (macOS)
+      if: startsWith(matrix.os, 'macos')
+      run: brew install check meson
```
Comment (hackermd) [inline .github/workflows/run_unit_tests.yml:44]: Should we install `uthash` via the package managers now?

File: include/dicom.h [hunk 3]
```diff
@@ -5,6 +5,21 @@
...
 #define DCM_INCLUDED
 
+#if defined(_WIN32) && !defined(__GNUC__)
+#include <BaseTsd.h>
+typedef SSIZE_T ssize_t;
+#endif
+
+#ifdef _WIN32
+#ifdef BUILDING_LIBDICOM
+#define DICOM_EXTERN __declspec(dllexport) extern
```
Comment (hackermd) [inline include/dicom.h:15]: Do we want to use `DCM_` rather than `DICOM_` or does this have to match the name of the library?

**Comment by bgilbert** [comment]
Thanks, updated!

**Review by hackermd — APPROVED** [review]

File: .github/workflows/run_unit_tests.yml (hunk 1 above)
Comment (ninjaoflight) [reply .github/workflows/run_unit_tests.yml:51]: Hi, another known workaround is to set both `clang++` and `lld-link` to `CXX` and `CXX_LLD` environment variables so you don´t have issues with the link issue while using a newer meson version like 1.0.0, thanks for the notice on this issue I will add it on the list of known workarounds

Comment (bgilbert) [reply .github/workflows/run_unit_tests.yml:51]: This is a C project, so `CXX` and `CXX_LD` are ignored.  Setting `CC` to a C++ compiler fails at setup time, but setting only `CC_LD` works.  PR in https://github.com/hackermd/libdicom/pull/29; thanks for the pointer.

Comment (ninjaoflight) [reply .github/workflows/run_unit_tests.yml:51]: right, mb, in that case as you mentioned `cc` should be `clang` but only `CC_LD` is required as you mentioned

**Pull request merged**